src/odemis/driver/smaract.py

```diff
@@ -1814,7 +1900,12 @@ def SA_MC_GetProperty_s(self, id, property_key, val, ioArraySize):
 
     def SA_MC_WaitForEvent(self, id, p_ev, timeout):
         ev = _deref(p_ev, SA_MC_Event)
-        time.sleep(0.25)
+        start_time = time.time()
+        
+        while time.time() < self._current_move_finish \
+            and time.time() < start_time + timeout.value:
+            time.sleep(0.05)
```

Comment:
I don't think it behaves properly if there is a timeout passed. It will conclude that the movement is completed. Maybe with the rest of the driver it's fine... but if it's not breaking other things, I'd write this loop like this:
``` python
while time.time() < self._current_move_finish:
     if  time.time() < start_time + timeout.value:
         return    # Special timeout value?
     time.sleep(0.05)
```